p2p/src/net/mock/backend.rs

```diff
@@ -208,21 +238,30 @@ where
         }
     }
 
-    /// Send request to remote peer
+    /// Sends a request to the remote peer.
     async fn send_request(
         &mut self,
-        peer_id: &MockPeerId,
+        peer_id: MockPeerId,
         request: message::Request,
     ) -> crate::Result<MockRequestId> {
         let peer = self
             .peers
-            .get_mut(peer_id)
+            .get_mut(&peer_id)
             .ok_or(P2pError::PeerError(PeerError::PeerDoesntExist))?;
 
-        let (request_id, request) = self.request_mgr.make_request(peer_id, request)?;
-
+        let (request_id, request) = self.request_mgr.make_request(request)?;
         peer.tx.send(MockEvent::SendMessage(request)).await.map_err(P2pError::from)?;
 
+        let is_inserted = self.pending_responses.insert(request_id);
+        debug_assert!(is_inserted);
+
+        let timeout = self.p2p_config.request_timeout.clone().into();
+        let sender = self.timeouts_sender.clone();
+        tokio::spawn(async move {
+            time::sleep(Duration::from_secs(timeout)).await;
```

Comment:
Just to be clear: this is `tokio::time::sleep`, not `std::thread::sleep`. According to the documentation, "no work is performed while awaiting on the sleep future to complete".

Reply:
I know. But it's the wrong tool for the job. 

Btw, also no work is performed on sleeps because a yield happens. But it's still the wrong way. Timers can be optimized by OS schedulers in ways that are blocked by sleeping.

Reply:
> Timers can be optimized by OS schedulers in ways that are blocked by sleeping.

Tokio uses a timer under the hood of the `Sleep` feature.